### PR TITLE
BUG: read_csv raising for arrow engine and parse_dates

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Not sure if this is an actual regression, but ties a bit into the dtype backend and would be nice if this works, since this raises for every parse_date case with numpy dtype backend (and I want to advertise the engine in my pyarrow blog...)

### CODE REVIEW

File: pandas/io/parsers/base_parser.py [hunk 1]
```diff
@@ -1137,6 +1137,9 @@ def unpack_if_single_element(arg):
         return arg
 
     def converter(*date_cols, col: Hashable):
+        if len(date_cols) == 1 and date_cols[0].dtype.kind in "Mm":
```
Comment (mroeschke) [inline pandas/io/parsers/base_parser.py:1140]: Not sure if I understood your comment in the OP correctly, but does this also fix `parse_dates` for numpy dtypes too?

**Comment by lithomas1** [comment]
Can you share a MRE of the issue you're having?

I'm wondering if #50056 fixes/causes the issue.

File: pandas/io/parsers/base_parser.py (hunk 1 above)
Comment (phofl) [reply pandas/io/parsers/base_parser.py:1140]: Only NumPy dtypes as far as I know, we fixed this for Arrow dtypes a couple of weeks ago.

**Comment by phofl** [comment]
```
data = """a,b
2000-01-01 00:00:00,1
2000-01-01 00:00:01,1"""

result = pd.read_csv(StringIO(data), parse_dates=["a"], engine="pyarrow")
```

Nope, still raises after your PR is in

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Comment by phofl** [comment]
Will backport later today or tomorrow

**Comment by lithomas1** [comment]
Might want to consider backporting #50056 and #52087 too.

**Comment by phofl** [comment]
The first one is a bit less common, can think about the second one if you like

**Comment by lithomas1** [comment]
That'd be great, I think there were a couple issues opened about nans in string columns not being read properly related to the second.

**Pull request merged**